### ISSUE TITLE
Question about KQAPro

### ISSUE THREAD

**Issue by mug2mag** [Flitternie/GraphQ_IR#6]
Great job! And I have a question:
When you use KQAPro, how do you get these five different divisions like "Multi-hop, Qualifier, Comparison, Logical, Count, Verify, Zero-shot"?

**Comment by mug2mag** [comment]
And did you test KQAPro on val.json or test.json?

**Comment by Flitternie** [comment]
Thanks for the questions. For the evaluation of KQA Pro, the accuracies we reported in the paper (Table 1) were tested on `test.json`. `val.json` was used as the validation data for checkpoint selection. 

For your first question, we used a Python script for differentiating KQA Pro query categories based on the operators contained in the corresponding KoPL queries. For details, please refer to the code in `data/kqapro/evaluate.py`.

**Comment by mug2mag** [comment]
@Flitternie 
To clarify, there are no answers in 'test.json' (only questions and their choices), so how do you get the result? If I missed something?

**Comment by mug2mag** [comment]
@Flitternie 
Sorry to bother you again.
I see that you split data " based on the logical form length and test the parsers to generate long queries based on the short query components seen in the training data" in the paper.  But I don't understand how the transformation happens that you can get class data "Qualifier, Comparison, Logical" in table 4.

**Comment by Flitternie** [comment]
> @Flitternie To clarify, there are no answers in `test.json` (only questions and their choices), so how do you get the result? I don't know if I missed something?

The ground truth logical forms of KQA Pro's test set are not publicly released. You need to submit your inference results to their leaderboard in order to obtain the evaluation results. Please refer to their official webpage for details: [http://thukeg.gitee.io/kqa-pro/leaderboard.html](http://thukeg.gitee.io/kqa-pro/leaderboard.html). 


> @Flitternie Sorry to bother you again. I see that you split data " based on the logical form length and test the parsers to generate long queries based on the short query components seen in the training data" in the paper. But I don't understand how the transformation happens that you can get class data "Qualifier, Comparison, Logical" in table 4.

In our paper, "Table 4" should be the experiment results on MetaQA-Cypher so I guess you are actually referring to **Table 8** in the Appendix. We used the same script `data/kqapro/evaluate.py` to categorize those queries.

**Comment by mug2mag** [comment]
Thanks very much for your reply with patience.
